### DISCUSSION TITLE
Schema file 'schema/schema.json' not found while trying to validate.

### DISCUSSION THREAD

**Discussion by awalter-transcarent** [CMSgov/price-transparency-guide-validator#59] [Q&A] [answered]
Hello,

I was able to download and install all required dependencies and I can successfully call the help and update commands from cms-mrf-validator:
![image](https://user-images.githubusercontent.com/101288111/178005463-868db3a1-6102-4126-988a-e83a19f81e04.png)

When trying to run the validate command on any file, I run into this error: "Schema file 'schema/schema.json' not found"
![image](https://user-images.githubusercontent.com/101288111/178005816-6d321532-f79d-4b37-b684-420670ceddc7.png)

I don't know C++ or Node JS but there is no schema/schema.json in the repo so I looked through the codebase. I believe it may be an issue with a temporary file that is created. 

I found the print command here: https://github.com/CMSgov/price-transparency-guide-validator/blob/main/schemavalidator.cpp#:~:text=fprintf(outFile%2C%20%22Schema%20file%20%27%25s%27%20not%20found%5Cn%22%2C%20schemaPath.c_str())%3B

Any help would be appreciated.

**Comment by olderthandirt5** [comment]
I have run into the same issue running on Ubuntu. Can run help and update commands, but validate command generates error:
Schema file 'schema/schema.json' not found

Build generated no errors:
sudo docker build -t validator .
Sending build context to Docker daemon  4.378MB
Step 1/9 : FROM ubuntu
 ---> 27941809078c
Step 2/9 : ARG VERSION=v1.0.0
 ---> Using cache
 ---> 264723f6829d
Step 3/9 : RUN apt-get update
 ---> Using cache
 ---> 3ddb0e724d9f
Step 4/9 : RUN apt-get install -y g++ cmake doxygen valgrind wget
 ---> Using cache
 ---> 120cf277aef2
Step 5/9 : COPY ./schemavalidator.cpp /
 ---> Using cache
 ---> e14a1442fdc0
Step 6/9 : COPY ./rapidjson /rapidjson
 ---> Using cache
 ---> 9948c4c6195f
Step 7/9 : COPY ./tclap /tclap
 ---> Using cache
 ---> cff30e372ea0
Step 8/9 : RUN g++ -I /tclap/include/ schemavalidator.cpp -o validator
 ---> Using cache
 ---> bbdaee910066
Step 9/9 : ENTRYPOINT ["/validator"]
 ---> Using cache
 ---> d6ee4b987f2e
Successfully built d6ee4b987f2e
Successfully tagged validator:latest

**Comment by olderthandirt5** [comment]
FYI same code runs fine on Windows, even after updating schemas.

@shaselton-usds  Is this related to issue #60?

**Reply by shaselton-usds** [reply]
@olderthandirt5 I don't think it is. I think the problem might be that the "update" command hasn't been ran prior to trying to run the validator, but still trying to confirm this.

cc @mint-thompson

**Reply by olderthandirt5** [reply]
In our environment, I have run update and then the validator on Windows and Ubuntu. It works fine on Windows and throws the error on Ubuntu. On Ubuntu, I use 'sudo' commands to run the update and validate commands.

Is there anything that needs to be done differently on the Linux environment to install the program?

![image](https://user-images.githubusercontent.com/100310916/179571671-445d7130-5ae3-49f2-b962-1f6429427e54.png)

**Reply by mint-thompson** [reply]
This is very strange. Could you try rebuilding the docker image in your Ubuntu environment?

**Comment by olderthandirt5** [comment]
I rebuilt it in a new directory, successfully updated the schemas, but got the same error.
Here is the Docker version :
Docker version 20.10.14, build a224086349

**Comment by shaselton-usds** [comment]
@olderthandirt5 #60 was merged in -- let me know if you are still running into issues.

**Reply by rahulpkomodo** [reply]
I just tried and also running into this issue. I am running on Mac OS X

**Comment by olderthandirt5** [comment]
I am still getting the same error as before. I am going to try a couple of other things, but if you have any testing you would like me to do, I am willing.

**Comment by olderthandirt5** [comment, marked as answer]
As an 'experienced' developer, I am best described by Oscar Wilde:
"Experience is simply the name we give our mistakes."

In my case, I am fairly certain I created this problem during the install of the cms-mrf-validator on our instance of Ubuntu 22.04.1 by not following the instructions to the letter. This was a new VM and I had installed several applications to get the environment ready. When I got to the line in the instructions where it said to verify that docker was running by typing 'docker ps', I must have typed 'sudo docker ps'. 

Today I realized the problem was a mismatch in the gid of  /var/run/docker.sock and the docker group, which did not exist.  I tried many things, some of which helped and some which did not, but I think for anyone running into this issue I would recommend

1. Ensure that running 'docker ps' works with a non-privileged user.
2. Delete any previous installations of the validator.
3. Reinstall by following the instructions exactly.

After doing this, I was able to run the validator against a large pricing file with no errors.

**Comment by olderthandirt5** [comment]
I am still having issues with this problem, even after making sure the permissions were the same between the docker group and docker.sock (on Ubuntu 22.4). Is anyone else still getting this error?
